src/main/java/com/rog/EShop/entity/User.java

```diff
@@ -1,38 +1,32 @@
 package com.rog.EShop.entity;
 
-import org.springframework.security.core.userdetails.UserDetails;
-
 import javax.persistence.*;
-import java.io.Serializable;
 import java.time.LocalDateTime;
-import java.util.Collection;
 import java.util.List;
 import java.util.Objects;
+import java.util.UUID;
 
 @Entity
 @Table(name = "USERS")
-public class User implements Serializable, UserDetails {
+public class User {
     @Id
     @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "users_id_generator")
     @SequenceGenerator(name = "users_id_generator", sequenceName = "user_id_seq", allocationSize = 1)
     private Long id;
+    private UUID keycloakId;
     private String username;
-    private String password;
+    private String email;
     private String firstName;
     private String lastName;
     private LocalDateTime registerDate;
-    private Boolean accountNonExpired;
-    private Boolean accountNonLocked;
-    private Boolean credentialsNonExpired;
-    private Boolean enabled;
-    @ElementCollection(targetClass = Role.class)
-    @CollectionTable(name = "users_roles", joinColumns = @JoinColumn(name = "user_id"))
-    @Column(name = "role")
-    @Enumerated(EnumType.STRING)
+    @Transient
     private List<Role> roles;
```

Comment:
remove field `@Transient private List<Role> roles;` at all, we don't use it, and will not use